**MethaDory relies on various previously published packages and datasets:**

*Missing values imputation*

- Plaksienko, A., Di Lena, P., Nardini, C., & Angelini, C. (2024). [methyLImp2: faster missing value estimation for DNA methylation data](https://doi.org/10.1093/bioinformatics/btae001). Bioinformatics (Oxford, England), 40(1), btae001. 


*Cell proportion deconvolution*

- Teschendorff, A. E., Breeze, C. E., Zheng, S. C., & Beck, S. (2017). [A comparison of reference-based algorithms for correcting cell-type heterogeneity in Epigenome-Wide Association Studies](https://doi.org/10.1186/s12859-017-1511-5). BMC bioinformatics, 18(1), 105.  


*Chromosomal sex and age prediction*

- Pidsley, R., Y Wong, C. C., Volta, M., Lunnon, K., Mill, J., & Schalkwyk, L. C. (2013). [A data-driven approach to preprocessing Illumina 450K methylation array data](https://doi.org/10.1186/1471-2164-14-293). BMC genomics, 14, 293. 

*Dimensionality reduction*

- Blighe K., Lun A. (2024). [PCAtools: PCAtools: Everything Principal Components Analysis](https://github.com/kevinblighe/PCAtools). 

- Krijthe J., van der Maaten L. (2023). [T-Distributed Stochastic Neighbor Embedding using a Barnes-Hut Implementation](https://github.com/jkrijthe/Rtsne). 

- Konopka T., (2023) [Uniform Manifold Approximation and Projection](https://github.com/tkonopka/umap)
